peachjam/analysis/citations.py

```diff
@@ -0,0 +1,59 @@
+import shutil
+import tempfile
+
+import lxml.html
+from docpipe.pdf import pdf_to_text
+
+from peachjam.models import CitationLink
+
+
+class CitationAnalyser:
+    matchers = []
+
+    def extract_citations(self, document):
+        """Run matchers across the HTML or text in this document."""
+        if document.content_html_is_akn:
+            # don't markup AKN HTML
+            return False
+
+        if document.content_html:
+            # markup html
+            return self.extract_citations_from_html(document)
+        else:
+            # markup the source file by extracting the text and assuming we'll render it as a PDF
+            return self.extract_citations_from_source_file(document)
+
+    def extract_citations_from_html(self, document):
+        html = lxml.html.fromstring(document.content_html)
+        for matcher in self.matchers:
+            matcher().markup_html_matches(document.expression_uri(), html)
+        document.content_html = lxml.html.tostring(html, encoding="unicode")
```

Comment:
What's the reason behind not storing these citation links like when they are extracted from the source document in `extract_citations_from_source_file()` below?

Reply:
HTML citations are marked up in the html itself, as `a` elements, and therefore HTML extractions don't need the CitationLinks. We can't do that for PDF, so they're stored as CitationLinks.